tools/opscan/cmd/main.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/node"
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/disperser/dataapi"
+	"github.com/Layr-Labs/eigenda/disperser/dataapi/subgraph"
+	"github.com/Layr-Labs/eigenda/tools/opscan"
+	"github.com/Layr-Labs/eigenda/tools/opscan/flags"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/urfave/cli"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var (
+	version   = ""
+	gitCommit = ""
+	gitDate   = ""
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Version = fmt.Sprintf("%s,%s,%s", version, gitCommit, gitDate)
+	app.Name = "opscan"
+	app.Description = "operator network scanner"
+	app.Usage = ""
+	app.Flags = flags.Flags
+	app.Action = RunScan
+	if err := app.Run(os.Args); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func RunScan(ctx *cli.Context) error {
+	config, err := opscan.NewConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	logger, err := common.NewLogger(config.LoggerConfig)
+	if err != nil {
+		return err
+	}
+
+	subgraphApi := subgraph.NewApi(config.SubgraphEndpoint, config.SubgraphEndpoint)
+	subgraphClient := dataapi.NewSubgraphClient(subgraphApi, logger)
+
+	semvers := make(map[string]int)
+	if config.OperatorId != "" {
+		operatorInfo, err := subgraphClient.QueryOperatorInfoByOperatorId(context.Background(), config.OperatorId)
+		if err != nil {
+			logger.Warn("failed to fetch operator info", "operatorId", config.OperatorId, "error", err)
+			return errors.New("operator info not found")
+		}
+
+		operatorSocket := core.OperatorSocket(operatorInfo.Socket)
+		retrievalSocket := operatorSocket.GetRetrievalSocket()
+		semver := getNodeInfo(context.Background(), retrievalSocket, config.Timeout, logger)
+		semvers[semver]++
+
+	} else {
+		indexedOperatorState, err := subgraphClient.QueryIndexedOperatorsWithStateForTimeWindow(context.Background(), 90, dataapi.Registered)
```

Comment:
TODO: This query is not returning the full/accurate operator set. Only returns 94 operators, but testnet has 132. Increasing the time window doesn't fix it. Will investigate further.